src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,219 @@
+package org.minions.devfund.noemiguzman.battleship;
+
+
+import java.util.Random;
+
+/**
+ * Ship class.
+ */
+public abstract class Ship {
+    private int length;
+    private boolean horizontal;
+    private int bowRow;
+    private int bowColumn;
+    private static final int SIZE_OCEAN = 20;
+    private boolean[] hit;
+
+    /**
+     * This is an abstract method and therefore has no body.
+     * @return ship string
+     */
+    public abstract String getShipType();
+
+
+    /**
+     * Returns true if it is okay to put a ship of this length with its bow in this location,
+     * with the given orientation, and returns false otherwise.
+     * The ship must not overlap another ship, or touch another ship (vertically, horizontally, or diagonally),
+     * and it must not stickout beyond the array.
+     * Do not actually change either the ship or the Ocean, just says whether it is legal to do so.
+     * @param row numeric
+     * @param column numeric
+     * @param horizontal true
+     * @param ocean object
+     * @return true
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 32 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)